modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -0,0 +1,671 @@
+	 * @param bool   $calypso Optional. Whether links should point to Calypso or wp-admin. Default true (Calypso).
+	 */
+	public function add_my_home_menu( $domain, $calypso = true ) {
+		global $submenu;
+
+		$menu_slug = $calypso ? 'https://wordpress.com/home/' . $domain : 'index.php';
+
+		remove_menu_page( 'index.php' );
+		remove_submenu_page( 'index.php', 'index.php' );
+
+		add_menu_page( __( 'My Home', 'jetpack' ), __( 'My Home', 'jetpack' ), 'read', $menu_slug, null, 'dashicons-admin-home', 2 );
+
+		// Only add submenu when there are other submenu items.
+		if ( ! empty( $submenu['index.php'] ) ) {
+			add_submenu_page( $menu_slug, __( 'My Home', 'jetpack' ), __( 'My Home', 'jetpack' ), 'read', $menu_slug, null, 1 );
+		}
+
+		$this->migrate_submenus( 'index.php', $menu_slug );
+	}
+
+	/**
+	 * Adds Stats menu.
+	 *
+	 * @param string $domain Site domain.
+	 */
+	public function add_stats_menu( $domain ) {
+		$menu_title = __( 'Stats', 'jetpack' );
+
+		if ( $this->is_wpcom_site() && ! $this->is_api_request ) {
+			$menu_title .= sprintf(
+				'<img class="sidebar-unified__sparkline" width="80" height="20" src="%1$s" alt="%2$s">',
+				esc_url( home_url( 'wp-includes/charts/admin-bar-hours-scale-2x.php?masterbar=1&s=' . get_current_blog_id() ) ),
+				esc_attr__( 'Hourly views', 'jetpack' )
+			);
+		}
+
+		add_menu_page( __( 'Stats', 'jetpack' ), $menu_title, 'edit_posts', 'https://wordpress.com/stats/day/' . $domain, null, 'dashicons-chart-bar', 3 );
+	}
+
+	/**
+	 * Adds Purchases menu.
+	 *
+	 * @param string $domain Site domain.
+	 */
+	public function add_purchases_menu( $domain ) {
+		remove_menu_page( 'paid-upgrades.php' );
+		add_menu_page( __( 'Purchases', 'jetpack' ), __( 'Purchases', 'jetpack' ), 'manage_options', 'https://wordpress.com/plans/' . $domain, null, 'dashicons-cart', 4 );
+		$this->migrate_submenus( 'paid-upgrades.php', 'https://wordpress.com/plans/' . $domain );
+	}
+
+	/**
+	 * Adds Posts menu.
+	 *
+	 * @param string $domain  Site domain.
+	 * @param bool   $calypso Optional. Whether links should point to Calypso or wp-admin. Default true (Calypso).
+	 */
+	public function add_posts_menu( $domain, $calypso = true ) {
+		if ( ! $calypso ) {
+			return;
+		}
+
+		$ptype_obj = get_post_type_object( 'post' );
+		$menu_slug = 'https://wordpress.com/posts/' . $domain;
+
+		remove_menu_page( 'edit.php' );
+		remove_submenu_page( 'edit.php', 'edit.php' );
+		remove_submenu_page( 'edit.php', 'post-new.php' );
+
+		add_menu_page( esc_attr( $ptype_obj->labels->menu_name ), $ptype_obj->labels->menu_name, $ptype_obj->cap->edit_posts, $menu_slug, null, 'dashicons-admin-post', $ptype_obj->menu_position );
+		add_submenu_page( $menu_slug, $ptype_obj->labels->all_items, $ptype_obj->labels->all_items, $ptype_obj->cap->edit_posts, $menu_slug, null, 5 );
+		add_submenu_page( $menu_slug, $ptype_obj->labels->add_new, $ptype_obj->labels->add_new, $ptype_obj->cap->create_posts, 'https://wordpress.com/post/' . $domain, null, 10 );
```

Comment:
I think there are situations where we'd want to point to wp-admin instead of Calypso to manage posts (and pages). Sites using the Classic Editor come to mind. Explorers, who worked on the Calypso editor deprecation, could tell us a bit more about this I think.

Reply:
[There's a toggle for that purpose that's currently hard-coded](https://github.com/Automattic/jetpack/blob/8d01344c2fedb2edf9f96ff45a3b83adabed48ac/modules/masterbar/admin-menu/class-admin-menu.php#L82-L88). Work on that is tracked in https://github.com/Automattic/wp-calypso/issues/47043